src/lib/storage/storage_manager.cpp

```diff
@@ -30,123 +34,149 @@ void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> t
   }
 
   // Create table statistics and chunk pruning statistics for added table.
+
   table->set_table_statistics(TableStatistics::from_table(*table));
   generate_chunk_pruning_statistics(table);
 
-  _tables.emplace(name, std::move(table));
+  if (table_iter == _tables.end())
+    _tables.emplace(name, std::move(table));
+  else
+    _tables[name] = std::move(table);
```

Comment:
Do we have a test that adds, drops, and adds again?

Reply:
Sure: https://github.com/hyrise/hyrise/pull/2072/files#diff-6a6d6b4d1981779e9817b29e0e581c5aR75